mm2src/coins/eth.rs

```diff
@@ -3954,6 +4217,19 @@ impl EthCoin {
         Ok(None)
     }
 
+    pub async fn get_watcher_reward_amount(&self) -> Result<BigDecimal, MmError<WatcherRewardError>> {
+        let gas_price = repeatable!(async { self.get_gas_price().compat().await.retry_on_err() })
+            .attempts(3)
```

Comment:
Should this number of attempts be increased to avoid unwanted swap failures? Should we rely on a timeout instead? `maker_payment_wait`/`wait_taker_payment` have been used before in similar situations, can they be used here instead or should we choose different timeouts?

Reply:
Yes it makes sense to repeat the gas price function calls within the defined swap durations, I added timeouts for this. We should also note that in other parts of the project the gas price function calls are done only once and swap fails if this call fails.